lib/rubocop/cop/lint/deprecated_constants.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for deprecated constants.
+      #
+      # It has `DeprecatedConstants` option. If there is an alternative method, you can set the
+      # you can set alternative value as the first element of value. Or set `null` to the value
+      # if there is no alternative value.
+      # And you can specify the deprecated version as the second element of value.
+      # Or set `null` to the value if there is unknown deprecated version.
+      #
+      #   DeprecatedConstants:
+      #     'DEPRECATED_CONSTANT': ['alternative_value', 'deprecated_version']
+      #
+      # By default, `NIL`, `TRUE`, `FALSE` and `Random::DEFAULT` are configured.
+      #
+      # @example
+      #
+      #   # bad
+      #   NIL
+      #   TRUE
+      #   FALSE
+      #   Random::DEFAULT # Return value of Ruby 2 is `Random` instance, Ruby 3.0 is `Random` class.
+      #
+      #   # good
+      #   nil
+      #   true
+      #   false
+      #   Random.new # `::DEFAULT` has been deprecated in Ruby 3, `.new` is compatible with Ruby 2.
+      #
+      class DeprecatedConstants < Base
+        extend AutoCorrector
+
+        SUGGEST_GOOD_MSG = 'Use `%<good>s` instead of `%<bad>s`%<deprecated_message>s.'
+        DO_NOT_USE_MSG = 'Do not use `%<bad>s`%<deprecated_message>s.'
+
+        def on_const(node)
+          constant = node.absolute? ? consntant_name(node, node.short_name.to_s) : node.source
+          return unless deprecated_constants.key?(constant)
+
+          good, deprecated_version = deprecated_constants[constant]
+
+          add_offense(node, message: message(good, node.source, deprecated_version)) do |corrector|
+            corrector.replace(node, good)
+          end
+        end
+
+        private
+
+        def consntant_name(node, nested_constant_name)
+          return nested_constant_name unless node.namespace.const_type?
+
+          consntant_name(node.namespace, "#{node.namespace.short_name}::#{nested_constant_name}")
+        end
+
+        def message(good, bad, deprecated_version)
+          if deprecated_version
+            deprecated_message = ", which is deprecated in Ruby #{deprecated_version}"
```

Comment:
"deprecated since" would be a slightly better wording here

Reply:
I see! Thank your for the suggestion!